cbits/iterator.c

```diff
@@ -0,0 +1,24 @@
+#include <HsFFI.h>
+#include <jni.h>
+
+#ifdef __cplusplus
```

Comment:
This is unnecessary boilerplate. See http://stackoverflow.com/questions/3789340/combining-c-and-c-how-does-ifdef-cplusplus-work